--- boost_1_68_0/libs/context/build/Jamfile.v2.orig	2018-08-01 15:50:46.000000000 -0500
+++ boost_1_68_0/libs/context/build/Jamfile.v2	2018-09-06 15:08:03.000000000 -0500
@@ -460,6 +460,17 @@
      <address-model>32
      <architecture>x86
      <binary-format>elf
+     <toolset>cray
+   ;
+
+alias asm_sources
+   : asm/make_i386_sysv_elf_gas.S
+     asm/jump_i386_sysv_elf_gas.S
+     asm/ontop_i386_sysv_elf_gas.S
+   : <abi>sysv
+     <address-model>32
+     <architecture>x86
+     <binary-format>elf
      <toolset>gcc
    ;
 
@@ -579,6 +590,17 @@
      <address-model>64
      <architecture>x86
      <binary-format>elf
+     <toolset>cray
+   ;
+
+alias asm_sources
+   : asm/make_x86_64_sysv_elf_gas.S
+     asm/jump_x86_64_sysv_elf_gas.S
+     asm/ontop_x86_64_sysv_elf_gas.S
+   : <abi>sysv
+     <address-model>64
+     <architecture>x86
+     <binary-format>elf
      <toolset>gcc
    ;
 
@@ -701,6 +723,17 @@
    ;
 
 alias asm_sources
+   : asm/make_x86_64_sysv_elf_gas.S
+     asm/jump_x86_64_sysv_elf_gas.S
+     asm/ontop_x86_64_sysv_elf_gas.S
+   : <abi>x32
+     <address-model>64
+     <architecture>x86
+     <binary-format>elf
+     <toolset>cray
+   ;
+
+alias asm_sources
    : asm/make_x86_64_sysv_elf_gas.S
      asm/jump_x86_64_sysv_elf_gas.S
      asm/ontop_x86_64_sysv_elf_gas.S
