diff --git a/packages/amesos/src/Amesos_Superludist.cpp b/packages/amesos/src/Amesos_Superludist.cpp
index f0542fa..0672848 100644
--- a/packages/amesos/src/Amesos_Superludist.cpp
+++ b/packages/amesos/src/Amesos_Superludist.cpp
@@ -53,7 +53,7 @@ public:
   //! SuperLU_DIST's grid information.
   gridinfo_t grid_;
   //! Vector of options.
-#if SUPERLU_DIST_MAJOR_VERSION == 5
+#if SUPERLU_DIST_MAJOR_VERSION >= 5
   //Note we may add the need for minor or patch version as need
   superlu_dist_options_t options_;
 #else
@@ -472,7 +472,13 @@ int Amesos_Superludist::Factor()
     }
 
     if( RowPerm_ == "NATURAL" ) PrivateSuperluData_->options_.RowPerm = (rowperm_t)NATURAL;
-    if( RowPerm_ == "LargeDiag" ) PrivateSuperluData_->options_.RowPerm = LargeDiag;
+    if( RowPerm_ == "LargeDiag" || RowPerm_ == "LargeDiag_MC64" )
+#if (SUPERLU_DIST_MAJOR_VERSION > 5) || (SUPERLU_DIST_MAJOR_VERSION == 5 && SUPERLU_DIST_MINOR_VERSION >= 4)
+      PrivateSuperluData_->options_.RowPerm = LargeDiag_MC64;
+    if( RowPerm_ == "LargeDiag_AWPM" ) PrivateSuperluData_->options_.RowPerm = LargeDiag_AWPM;
+#else
+      PrivateSuperluData_->options_.RowPerm = LargeDiag;
+#endif
     else if( ColPerm_ == "MY_PERMR" ) {
       PrivateSuperluData_->options_.RowPerm = MY_PERMR;
       PrivateSuperluData_->ScalePermstruct_.perm_r = perm_r_;
